app/scripts/modules/cloudfoundry/serverGroup/configure/serverGroupArtifactSettingsDirective.html

```diff
@@ -0,0 +1,54 @@
+<ng-form name="artifactSettings">
+  <div class="form-group">
+    <div class="col-md-3 sm-label-left">
+      <b>Repository</b>
+    </div>
+    <div class="col-md-7">
+      <input type="text"
+             class="form-control input-sm"
+             name="repository"
+             ng-model="command.repository"/>
+    </div>
+  </div>
+  <div class="form-group">
+    <div class="col-md-3 sm-label-left">
+      <b>Artifact</b>
+    </div>
+    <div class="col-md-7">
+      <input type="text"
+             class="form-control input-sm"
+             name="artifact"
+             ng-model="command.artifact"/>
+    </div>
+  </div>
+  <div class="form-group">
+    <div class="col-md-3 sm-label-left">
+      <b ng-if="!artifactSettingsCtrl.isS3()">Username</b>
+      <b ng-if="artifactSettingsCtrl.isS3()">Access Key</b>
+    </div>
+    <div class="col-md-7">
+      <input type="text"
+             class="form-control input-sm"
+             name="username"
+             ng-if="!artifactSettingsCtrl.isS3()"
+             ng-model="command.username"/>
+      <input type="password"
+             class="form-control input-sm"
+             name="username"
+             ng-if="artifactSettingsCtrl.isS3()"
+             ng-model="command.username"/>
+    </div>
+  </div>
+  <div class="form-group">
+    <div class="col-md-3 sm-label-left">
+      <b ng-if="!artifactSettingsCtrl.isS3()">Password</b>
+      <b ng-if="artifactSettingsCtrl.isS3()">Secret Key</b>
+    </div>
+    <div class="col-md-7">
+      <input type="password"
+             class="form-control input-sm"
+             name="password"
+             ng-model="command.password"/>
+    </div>
+  </div>
```

Comment:
You can remove a lot of the boilerplate by using the `stageConfigField` directive, e.g.
```html
<div class="form-group">
  <div class="col-md-3 sm-label-left">
    <b>Artifact</b>
  </div>
  <div class="col-md-7">
    <input type="text"
           class="form-control input-sm"
           name="artifact"
           ng-model="command.artifact"/>
  </div>
</div>
```
becomes
```html
<stage-config-field label="Artifact">
  <input type="text"
         class="form-control input-sm"
         name="artifact"
         ng-model="command.artifact"/>
</stage-config-field>
```